elasticgraph-warehouse/spec/unit/elastic_graph/warehouse/schema_definition/warehouse_table_spec.rb

```diff
@@ -0,0 +1,106 @@
+# Copyright 2024 - 2025 Block, Inc.
+#
+# Use of this source code is governed by an MIT-style
+# license that can be found in the LICENSE file or at
+# https://opensource.org/licenses/MIT.
+#
+# frozen_string_literal: true
+
+require "elastic_graph/warehouse/schema_definition/api_extension"
+
+module ElasticGraph
+  module Warehouse
+    module SchemaDefinition
+      RSpec.describe WarehouseTable, :warehouse_schema do
+        it "generates table schema with warehouse_table definition" do
+          results = define_warehouse_schema do |s|
+            s.object_type "Product" do |t|
+              t.field "id", "ID"
+              t.field "name", "String"
+              t.field "price", "Float"
+              t.warehouse_table "products"
+            end
+          end
+
+          product_type = results.state.object_types_by_name.fetch("Product")
+          table = product_type.warehouse_table_def
+
+          expect(table).to be_a(WarehouseTable)
+          expect(table.name).to eq("products")
+
+          expect(table.indexed_type).to eq(product_type)
+        end
+
+        it "converts table to configuration hash" do
+          results = define_warehouse_schema do |s|
+            s.object_type "Order" do |t|
+              t.field "order_id", "ID"
+              t.field "total", "Float"
+              t.warehouse_table "orders"
+            end
+          end
+
+          order_type = results.state.object_types_by_name.fetch("Order")
+          table = order_type.warehouse_table_def
+
+          config = table.to_config
+          expect(config).to have_key("table_schema")
+          expect(config["table_schema"]).to include("CREATE TABLE IF NOT EXISTS orders")
+          expect(config["table_schema"]).to include("order_id STRING")
+          expect(config["table_schema"]).to include("total DOUBLE")
```

Comment:
Similar to my comment elsewhere, I'd prefer to see this assert on the whole structure.